app/queries/articles/api_search_query.rb

```diff
@@ -0,0 +1,50 @@
+module Articles
+  class ApiSearchQuery
+    DEFAULT_PER_PAGE = 30
+
+    def self.call(...)
+      new(...).call
+    end
+
+    def initialize(params)
+      @q = params[:q]
+      @top = params[:top]
+      @page = params[:page].to_i
+      @per_page = [(params[:per_page] || DEFAULT_PER_PAGE).to_i, per_page_max].min
+    end
+
+    def call
+      @articles = published_articles_with_users_and_organizations
+
+      if q.present?
+        @articles = query_articles
+      end
+
+      if top.present?
+        @articles = top_articles.order(public_reactions_count: :desc)
+      end
+
+      @articles.page(page).per(per_page || DEFAULT_PER_PAGE)
```

Comment:
There's already a per_page logic in place on line 13. Is the extra `per(per_page || DEFAULT_PER_PAGE)` here necessary?